types/enigma.js/index.d.ts

```diff
@@ -0,0 +1,196 @@
+// Type definitions for enigma.js 2.2
+// Project: https://github.com/qlik-oss/enigma.js/
+// Definitions by: Konrad Mattheis <https://github.com/konne>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+declare module "enigma.js" {
+    const e: IEnigmaJSClass;
+    export = e;
+}
+
+interface IEnigmaJSClass {
+    /**
+     * Create a session object.
+     * @returns - Returns a session.
+     * Note: See Configuration for the configuration options.
+     */
+    create(config: enigmaJS.IConfig): enigmaJS.ISession;
+}
+
+declare namespace enigmaJS {
+    type MixinType = "Doc"| "GenericObject"| "GenericBookmark" | string;
+
+    interface IMixin {
+        /**
+         * QIX Engine types like for example GenericObject, Doc, GenericBookmark, are supported but also custom GenericObject
+         * types such as barchart, story and myCustomType.
+         * An API will get both their generic type as well as custom type mixins applied.
+         */
+        types: [MixinType];
```

Comment:
Is this supposed to be an array, or a singleton tuple? Because this is singleton tuple syntax.

Reply:
@DanielRosenwasser  thx, this have to be an array, I will fix it.